### PR TITLE
Simplify event callback execution filters

### LINKED ISSUE #14556: Remove null wildcard matching from event callback execution
## Summary

Remove null wildcard matching from event callback execution lookups and stop constructing callbacks with null routing fields.

## Problem

`SQLEventCallbackService.execute_callbacks()` previously fetched callbacks with predicates shaped like:

```python
or_(
    StoredEventCallback.event_kind == event.kind,
    StoredEventCallback.event_kind.is_(None),
)
or_(
    StoredEventCallback.conversation_id == conversation_id,
    StoredEventCallback.conversation_id.is_(None),
)
```

This treats `event_kind = NULL` as "all event kinds" and `conversation_id = NULL` as "all conversations". That makes the hot callback lookup broader and harder to optimize.

If callback execution no longer supports null wildcard semantics, the constructor/model should not keep accepting null values that create callbacks which will never execute.

The main production callback registration path is the title-setting callback, and that only handles `MessageEvent` in code. Instead of relying on `event_kind = NULL` as a wildcard and then returning early for non-message events, it should be registered explicitly for `MessageEvent`.

## Proposed change

Change `execute_callbacks()` so it only executes exact callback matches:

```python
StoredEventCallback.status == EventCallbackStatus.ACTIVE
StoredEventCallback.event_kind == event.kind
StoredEventCallback.conversation_id == conversation_id
```

Also update callback models/registration so new callbacks have concrete routing fields:

```python
conversation_id=<conversation uuid>
event_kind='MessageEvent'
```

## Acceptance criteria

- `execute_callbacks()` no longer includes `event_kind IS NULL` or `conversation_id IS NULL` wildcard branches.
- `CreateEventCallbackRequest` / `EventCallback` reject `conversation_id=None`.
- `CreateEventCallbackRequest` / `EventCallback` reject `event_kind=None`.
- New callbacks default or set `event_kind='MessageEvent'` where appropriate.
- A legacy row with `conversation_id = NULL` is not executed for arbitrary conversations.
- A legacy row with `event_kind = NULL` is not executed for arbitrary event kinds.
- Conversation-specific callbacks with exact event kind continue to execute normally.
- New `SetTitleCallbackProcessor` registrations use `event_kind='MessageEvent'`.
- Tests cover exact callback execution, model validation, and ignored legacy null conversation/event-kind rows.

---

_This issue was updated by an AI agent (OpenHands) on behalf of the user._

### PR DESCRIPTION
**Pull request by neubig**
HUMAN: Slightly reduce flexibility of our event callback queries to hopefully increase efficiency and decrease DB load.

- [ ] A human has tested these changes

## Summary

- Remove `event_kind IS NULL` and `conversation_id IS NULL` wildcard matching from `SQLEventCallbackService.execute_callbacks()`.
- Tighten `CreateEventCallbackRequest` / `EventCallback` so new callbacks require concrete `conversation_id` and non-null `event_kind`.
- Add `EventCallbackProcessor.get_event_kind()` and use it when registering startup processors.
- Register enterprise V1 integration processors for `ConversationStateUpdateEvent` and title callbacks for `MessageEvent`.
- Add tests for exact callback execution, model validation, and ignored legacy null conversation/event-kind rows.

Closes #14556

## Context

The callback execution query is on a hot event path and previously expanded each lookup with null wildcard branches. Callback processors already handle specific event types, so registrations now declare the event kind they actually handle and execution uses exact matching.

The DB columns remain nullable in this PR for backwards compatibility with existing rows; legacy null rows simply do not match execution anymore. A later migration can clean them up and add `NOT NULL` constraints/indexes if desired.

## Testing

- `OPENHANDS_SUPPRESS_BANNER=1 uv run python -m pytest tests/unit/app_server/test_sql_event_callback_service.py tests/unit/app_server/test_webhook_router_auto_title.py tests/unit/app_server/test_set_title_callback_processor.py -q`
- `OPENHANDS_SUPPRESS_BANNER=1 uv run ruff check openhands/app_server/event_callback/event_callback_models.py openhands/app_server/event_callback/sql_event_callback_service.py openhands/app_server/app_conversation/live_status_app_conversation_service.py openhands/app_server/event_callback/webhook_router.py tests/unit/app_server/test_sql_event_callback_service.py tests/unit/app_server/test_webhook_router_auto_title.py tests/unit/app_server/test_set_title_callback_processor.py enterprise/integrations/bitbucket/bitbucket_v1_callback_processor.py enterprise/integrations/bitbucket_data_center/bitbucket_dc_v1_callback_processor.py enterprise/integrations/github/github_v1_callback_processor.py enterprise/integrations/gitlab/gitlab_v1_callback_processor.py enterprise/integrations/jira/jira_v1_callback_processor.py enterprise/integrations/jira_dc/jira_dc_v1_callback_processor.py enterprise/integrations/slack/slack_v1_callback_processor.py`
- `cd enterprise && python3 -m pre_commit run --all-files --show-diff-on-failure --config ./dev_config/python/.pre-commit-config.yaml`

---

_This PR was updated by an AI agent (OpenHands) on behalf of the user._

---

To run this PR locally, use the following command:

GUI with Docker:
```
docker run -it --rm   -p 3000:3000   -v /var/run/docker.sock:/var/run/docker.sock   --add-host host.docker.internal:host-gateway   --name openhands-app-ed55b0b   docker.openhands.dev/openhands/openhands:ed55b0b
```

### CODE REVIEW

**Comment by neubig** [comment]
Addressed the review feedback in 3e55db4 by adding explicit callback processor event-kind declarations and registering enterprise V1 processors for `ConversationStateUpdateEvent`. Enterprise lint now passes locally.

_This comment was created by an AI agent (OpenHands) on behalf of the user._

File: openhands/app_server/app_conversation/live_status_app_conversation_service.py [hunk 1]
```diff
@@ -412,9 +412,15 @@ async def _start_app_conversation(
 
             # Save processors
             for processor in processors:
+                event_kind = (
+                    'MessageEvent'
+                    if isinstance(processor, SetTitleCallbackProcessor)
+                    else None
```
[the remark this answers is not in the document]
Comment (neubig) [reply openhands/app_server/app_conversation/live_status_app_conversation_service.py]: Fixed in 3e55db4: processors now declare an event kind via `EventCallbackProcessor.get_event_kind()`, and enterprise V1 processors are registered for `ConversationStateUpdateEvent`.

_This reply was created by an AI agent (OpenHands) on behalf of the user._

File: openhands/app_server/event_callback/sql_event_callback_service.py [hunk 2]
```diff
@@ -201,18 +201,8 @@ async def execute_callbacks(self, conversation_id: UUID, event: Event) -> None:
...
-                    StoredEventCallback.event_kind.is_(None),
-                )
-            )
-            .where(
-                or_(
-                    StoredEventCallback.conversation_id == conversation_id,
-                    StoredEventCallback.conversation_id.is_(None),
-                )
-            )
+            .where(StoredEventCallback.event_kind == event.kind)
```
[the remark this answers is not in the document]
Comment (neubig) [reply openhands/app_server/event_callback/sql_event_callback_service.py:204]: Confirmed/addressed in 3e55db4: startup processor registration now uses each processor’s explicit event kind, with enterprise V1 processors set to `ConversationStateUpdateEvent` and title callbacks set to `MessageEvent`.

_This reply was created by an AI agent (OpenHands) on behalf of the user._

File: tests/unit/app_server/test_sql_event_callback_service.py [hunk 3]
```diff
@@ -373,6 +376,83 @@ async def test_search_ordering(
...
+        assert len(callback_results) == 1
+        assert callback_results[0].event_callback_id == callback.id
+        assert callback_results[0].conversation_id == conversation_id
+
+    async def test_execute_callbacks_ignores_null_conversation_callback(
+        self,
+        service: SQLEventCallbackService,
+        sample_processor: EventCallbackProcessor,
+    ):
+        """Callbacks without a conversation_id should not execute globally."""
```
[the remark this answers is not in the document]
Comment (neubig) [reply tests/unit/app_server/test_sql_event_callback_service.py]: Thanks — leaving the legacy-null execution coverage in place while model validation now prevents new null-routed callbacks.

_This reply was created by an AI agent (OpenHands) on behalf of the user._

**Comment by neubig** [comment]
Addressed the latest bot suggestions in ed55b0b: `SetTitleCallbackProcessor` now declares `event_kind` explicitly and webhook registration uses `SetTitleCallbackProcessor.get_event_kind()` for consistency. Local unit tests, root Ruff, and enterprise lint all pass.

_This comment was created by an AI agent (OpenHands) on behalf of the user._

**Review by malhotra5 — APPROVED** [review]
LGTM
